tests/invalid_models_tests/test_ordinary_fields.py

```diff
@@ -1057,3 +1058,62 @@ class Meta:
 
         errors = Model._meta.get_field("field").check(databases=self.databases)
         self.assertEqual(errors, [])
+
+
+@isolate_apps("invalid_models_tests")
+class GeneratedFieldTests(TestCase):
+    @skipIfDBFeature("supports_generated_field")
+    def test_field_not_supported(self):
+        class Model(models.Model):
+            name = models.IntegerField()
+            field = models.GeneratedField(expression=F("name"))
+
+        self.assertEqual(
+            Model._meta.get_field("field").check(databases={"default"}),
+            [
+                Error(
+                    "%s does not support GeneratedFields." % connection.display_name,
+                    obj=Model._meta.get_field("field"),
+                    id="fields.E220",
+                ),
+            ],
+        )
```

Comment:
This should be checked for both cases:
```suggestion
    def test_field_not_supported(self):
        class Model(models.Model):
            name = models.IntegerField()
            field = models.GeneratedField(expression=F("name"))

        self.assertEqual(
            Model._meta.get_field("field").check(databases={"default"}),
            [
                Error(
                    "%s does not support GeneratedFields." % connection.display_name,
                    obj=Model._meta.get_field("field"),
                    id="fields.E220",
                ),
            ] if not connection.features.supports_generated_field else [],
        )
```
The same changes are needed for other feature flags. Also, we should check that `required_db_features` is respected (see for example `test_db_comment_required_db_features`).